src/auth/auth.service.ts

```diff
@@ -0,0 +1,64 @@
+import { Body, Injectable, UnauthorizedException } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { CreateUserDto } from 'src/users/dto/create-user.dto';
+import { UserLoginDto } from 'src/users/dto/user-login.dto';
+import { UserDto } from 'src/users/dto/user.dto';
+import { UsersService } from 'src/users/users.service';
+import { LoginStatus } from './interfaces/login-status.interface';
+import { JwtPayload } from './interfaces/payload.interface';
+import { RegistrationStatus } from './interfaces/registration-status.interface';
+
+@Injectable()
+export class AuthService {
+  constructor(
+    private readonly userService: UsersService,
+    private readonly jwtService: JwtService,
+  ) {}
+
+  async register(createUserDto: CreateUserDto): Promise<RegistrationStatus> {
+    let status: RegistrationStatus = {
+      success: true,
+      message: 'User registered successfully',
+    };
+
+    try {
+      await this.userService.create(createUserDto);
+    } catch (err) {
+      status = {
+        success: false,
+        message: err,
+      };
+    }
+
+    return status;
+  }
+
+  async login(userLoginDto: UserLoginDto): Promise<LoginStatus> {
+    const user = await this.userService.validateUserLogin(userLoginDto);
+
+    const token = this._createToken(user);
+
+    return {
+      username: user.username,
+      ...token,
```

Comment:
no need to spread token.
``return {
      username: user.username,
      token,
}``